pymc_marketing/mmm/lift_test.py

```diff
@@ -0,0 +1,533 @@
+"""Lift test functions for the MMM."""
+
+from functools import partial
+from typing import Callable, Optional, Union
+
+import numpy as np
+import pandas as pd
+import pymc as pm
+import pytensor.tensor as pt
+from numpy import typing as npt
+
+from pymc_marketing.mmm.transformers import logistic_saturation
+from pymc_marketing.mmm.utils import michaelis_menten
+
+
+class MissingLiftTestError(Exception):
+    def __init__(self, missing_values: npt.NDArray[np.int_]) -> None:
+        self.missing_values = missing_values
+        super().__init__(
+            f"Some lift test values are not in the model: {missing_values}"
+        )
+
+
+Index = npt.NDArray[np.int_]
+
+Values = Union[npt.NDArray[np.int_], npt.NDArray[np.float_], npt.NDArray[np.str_]]
+
+
+def _lift_test_index(lift_values: Values, model_values: Values) -> Index:
+    same_value = lift_values[:, None] == model_values
+    if not (same_value.sum(axis=1) == 1).all():
+        missing_values = np.argwhere(same_value.sum(axis=1) == 0).flatten()
+        raise MissingLiftTestError(missing_values)
+
+    return np.argmax(same_value, axis=1)
+
+
+Indices = dict[str, Index]
```

Comment:
maybe is better to have all the types together? so we move this up?

Reply:
Done!